src/main.rs

```diff
@@ -211,6 +211,33 @@ fn run_machete() -> anyhow::Result<bool> {
         }
     }
 
+    if has_unused_dependencies {
+        if !args.with_metadata {
+            println!("\n\
+                If you believe cargo-machete has detected an unused dependency incorrectly,\n\
+                you can try running it with the `--with-metadata` flag for better accuracy,\n\
+                though this may modify your Cargo.lock files.\n\
+                \n\
+                You can also add the dependency to the list of dependencies to ignore in the\n\
+                `[package.metadata.cargo-machete]` section of the appropriate Cargo.toml file.\n\
+                For example:\n\
+                \n\
+                [package.metadata.cargo-machete]\n\
+                ignored = [\"prost\"]\n"
+            );
+        } else {
+            println!("\n\
+                If you believe cargo-machete has detected an unused dependency incorrectly,\n\
```

Comment:
Could we use `const`ant str to avoid the code duplication here? This would also help spot that we're showing almost the same text, and that would highlight what the differences are too.

Reply:
I gave something like that a shot, but the result reads "inside-out", which is awkward:

```rust
    if has_unused_dependencies {
        let middle = if !args.with_metadata {
            "you can try running it with the `--with-metadata` flag for better accuracy,\n\
            though this may modify your Cargo.lock files.\n\
            \n\
            You can also add the dependency to the list of dependencies to ignore in the\n"
        } else {
            "you can add the dependency to the list of dependencies to ignore in the\n"
        };

        println!(
            "\n\
            If you believe cargo-machete has detected an unused dependency incorrectly,\n\
            {middle}\
            `[package.metadata.cargo-machete]` section of the appropriate Cargo.toml.\n\
            For example:\n\
            \n\
            [package.metadata.cargo-machete]\n\
            ignored = [\"prost\"]"
        );
    }
```

Maybe I'm missing an obvious way to make this less contorted.

In the end, I moved the part which is only applicable sometimes -- the `--with-metadata` suggestion -- to the end. That simplifies things, though it obviously means the `--with-metadata` part is mentioned second, which I originally didn't like as much. But it seems fine. Anyway, let me know what you prefer here.